wallet-next/src/worker.rs

```diff
@@ -64,12 +74,39 @@ impl Worker {
     }
 
     pub async fn run(mut self) -> Result<(), anyhow::Error> {
+        loop {
+            match self._run().await {
+                Ok(_) => {
+                    // If the worker returns `Ok` then it means it's done, so we can
+                    // stop looping.
+                    break;
+                }
+                Err(e) => {
+                    tracing::info!(?e, "wallet worker error");
+                    self.error_slot.lock().unwrap().replace(e);
+                }
+            };
+        }
+
+        Ok(())
+    }
+
+    async fn _run(&mut self) -> Result<(), anyhow::Error> {
```

Comment:
nit: I think it would be more consistent with other Tokio code to call this run_inner